x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/inference/results/RankedDocsResultsTests.java

```diff
@@ -37,6 +38,31 @@ public static RankedDocsResults.RankedDoc createRandomDoc() {
         return new RankedDocsResults.RankedDoc(randomIntBetween(0, 100), randomFloat(), randomBoolean() ? null : randomAlphaOfLength(10));
     }
 
+    public void test_asMap() {
+        var index = randomIntBetween(0, 100);
+        var score = randomFloat();
+        var docNullText = new RankedDocsResults.RankedDoc(index, randomFloat(), null);
+        var mapNullText = docNullText.asMap();
+        assert mapNullText.containsKey("index");
+        assert mapNullText.containsKey("relevanceScore");
+        assert mapNullText.containsKey("text");
+
+        assert mapNullText.get("index").equals(index);
+        assert mapNullText.get("relevanceScore").equals(score);
+        assert mapNullText.get("text").equals("Sample text");
+
+        var docWithText = new RankedDocsResults.RankedDoc(index, randomFloat(), "Sample text");
+        var mapWithText = docNullText.asMap();
+        assert mapNullText.containsKey("index");
+        assert mapNullText.containsKey("relevanceScore");
+        assert mapNullText.containsKey("text");
+
+        assert mapNullText.get("index").equals(index);
+        assert mapNullText.get("relevanceScore").equals(score);
+        assert mapNullText.get("text").equals("Sample text");
```

Comment:
You can simplify these assertions by using Hamcrest matchers

```suggestion
        var mapNullText = new RankedDocsResults.RankedDoc(index, score, null).asMap();
        assertThat(mapNullText, Matchers.is(Map.of("ranked_doc", Map.of("index", index, "relevance_score", score))));

        var mapWithText = new RankedDocsResults.RankedDoc(index, score, "Sample text").asMap();
        assertThat(mapWithText, Matchers.is(Map.of("ranked_doc", Map.of("index", index, "relevance_score", score, "text", "Sample text"))));
```